### ISSUE TITLE
How does association versioning work with `belongs_to`?

### ISSUE THREAD

**Issue by marnen** [airblade/paper_trail#503]
I have a use case where I have the following:

```ruby
class Project < ActiveRecord::Base
  has_paper_trail
  belongs_to :client
end
```
```ruby
class Client < ActiveRecord::Base
  has_paper_trail
  has_many :projects
end
```
I'd like to figure out how to do `@project.versions.last.reify` in such a way as to get the Client version along with the Project version. All the examples in the docs would have me do `@client.versions.last.reify has_many: true` and get a bunch of Project records, but that's not what I need: I just need one Project record, along with the Client version that was present when the Project version was saved. Is this possible (maybe `@project.versions.last.reify has_one: true`?), or do I just need to store the Client and Project version numbers separately?

**Comment by batter** [comment]
I believe this is a dup of #498.

Versioning of associations is incredibly complex, which is part of the reason why it took so long to even fathom working on this feature in the first place. As the [documentation regarding associations states](https://github.com/airblade/paper_trail#associations), this is not currently supported with the current source but if you have any ideas or suggestions then please give it a shot and open up a pull request.

**Comment by marnen** [comment]
I don't think this is a duplicate of #498, though it's certainly related. 

And yes, association versioning is hard (well, my own use case is simple, but it doesn't generalize).

**Comment by jaredbeck** [comment]
I noticed there hasn't been any activity on this in four months.  Marnen, is this still something you'd like to contribute?  Unless you're working on it, my preference would be to close the issue.  Thanks!

**Comment by theRealNG** [comment]
I have worked on this. 
https://github.com/theRealNG/paper_trail/commit/1257db3fe10fd2b8acbbc85c7fc7cb09506a8891

Will need to refactor the code a bit and write test cases. 
Will place a pull request once it is done, so please don't close this.

**Comment by jaredbeck** [comment]
> [I] will place a pull request ..

@theRealNG did this ever happen? I'd like to close this if no one is working on it.

**Comment by marnen** [comment]
@jaredbeck I haven't been actively working on the project for which I'm using paper_trail. But I confess I'm not sure why you'd close this: the issue still exists—even if no one is working on it, shouldn't it stay open so that people know it is still something that needs attention?

**Comment by theRealNG** [comment]
Sorry forgot about this. Finished working on it. Will work on writing the specs tomorrow.

**Comment by theRealNG** [comment]
@jared I think we can close this issue.

**Comment by jaredbeck** [comment]
Closed by https://github.com/airblade/paper_trail/pull/730. Thanks NG. :+1:
